test/inspector/test-inspector-port-zero-cluster.js

```diff
@@ -4,29 +4,58 @@
 const common = require('../common');
 const assert = require('assert');
 const cluster = require('cluster');
+const net = require('net');
 
 if (cluster.isMaster) {
-  const ports = [];
-  for (const worker of [cluster.fork(),
-                        cluster.fork(),
-                        cluster.fork()]) {
-    worker.on('message', common.mustCall((message) => {
-      ports.push(message.debugPort);
-      worker.kill();
-    }));
-    worker.send('debugPort');
-  }
-  process.on('exit', () => {
-    ports.sort();
-    assert.strictEqual(ports.length, 3);
-    assert(ports.every((port) => port > 0));
-    assert(ports.every((port) => port < 65536));
-    assert.strictEqual(ports[0] + 1, ports[1]);  // Ports should be consecutive.
-    assert.strictEqual(ports[1] + 1, ports[2]);
+  const oneFailSentinal = common.mustCall();
+  cluster.on('online', common.mustCall((worker) => worker.send('debugPort'), 2));
+  cluster.on('error', (err) => assert.fail(err));
+  cluster.on('exit', (worker, code, signal) => {
+    assert.ok(!signal);
+    // one fork should fail because of port clash with `server`
+    if (code === 12) return oneFailSentinal();
+    assert.strictEqual(0, code);
   });
+  const ports = [process.debugPort];
+  function serialFork() {
+    return new Promise((res) => {
+      const worker = cluster.fork();
+      // no common.mustCall since 1 out of 3 should fail
+      worker.on('online', () => {
+        worker.on('message', common.mustCall((message) => {
+          ports.push(message.debugPort);
+        }));
+      });
+      // since 1 out of 3 should fail
+      worker.on('disconnect', res);
+    });
+  }
+  
+  // block one of the ports with a listening socket
+  const server = net.createServer()
+  server.listen(process.debugPort + 2, common.localhostIPv4, common.mustCall(() => {
+    console.log(server.address());
+    // try to fork 3 workers No.2 should fail
+    Promise.all([serialFork(), serialFork(), serialFork()])
+      .then(common.mustCall(() => {
+        ports.sort();
+        // 3 = [master, worker1, worker3].length()
+        assert.strictEqual(ports.length, 3);
+        assert(ports.every((port) => port > 0));
+        assert(ports.every((port) => port < 65536));
+        // Ports should be consecutive.
+        assert.strictEqual(ports[0] + 1, ports[1]);
+        // But worker2 failed.
+        assert.strictEqual(ports[1] + 2, ports[2]);
+      })).catch((err) => console.error(err));
+  }));
+  server.unref();
 } else {
+  const sentinal = common.mustCall();
   process.on('message', (message) => {
-    if (message === 'debugPort')
-      process.send({ debugPort: process.debugPort });
+    if (message !== 'debugPort') return;
+    process.send({ debugPort: process.debugPort });
+    sentinal();
+    process.exit(0);
```

Comment:
Don't call process.exit(), call process.disconnect() and let the worker exit naturally.